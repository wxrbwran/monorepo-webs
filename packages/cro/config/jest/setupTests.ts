// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});
