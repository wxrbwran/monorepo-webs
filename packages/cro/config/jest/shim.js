import 'core-js/stable';
import 'regenerator-runtime/runtime';

/* eslint-disable import/first */
global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function requestAnimationFrame(callback) {
    setTimeout(callback, 0);
  };

global.matchMedia = window.matchMedia = () => ({
  matches: false,
  addListener: () => { },
  removeListener: () => { },
});
