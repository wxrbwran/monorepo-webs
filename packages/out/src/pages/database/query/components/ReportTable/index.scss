.report_wrap_table {
  // width: 1020px;
  width: 100%;
  transition: all 0.2s;
  &.full {
    width: 96vw;
    margin: 0 auto;
    position: fixed;
    left: 50%;
    top: 80px;
    transform: translateX(-50%);
    background-color: #fff;
    z-index: 2;
    min-height: 500px;
    .ant-table-wrapper {
      max-height: 80vh;
      overflow: auto;
    }
  }
  .table_top {
    @include flex(row, nowrap, space-between, center);
    margin-bottom: 15px;
    margin-top: 30px;
    &__info {
      font-size: 14px;
      color: #181818;
    }
    .btn-wrap {
      @include flex(row, nowrap, space-between, center);

      .create-btn {
        font-size: 14px;
        color: #3296FA;
        cursor: pointer;
      }
      .full-screen {
        margin: 0 15px;
        cursor: pointer;
        img {
          width: 24px;
          height: 24px;
        }
      }
    }
  }
  .table-export {
    margin-bottom: 12px;
    text-align: right;
    span {
      margin-left: 50px;
      font-size: 14px;
      color: #3296FA;
      cursor: pointer;
    }
  }
  .ant-table {
    border: 1px solid #f0f0f0;
    box-sizing: border-box;
  }

  .ant-table-cell {
    border-right: 1px solid #f0f0f0;
    box-sizing: border-box;
    position: relative;
    &:last-child {
      border-right: none;
    }
    .anticon-close {
      position: absolute;
      right: 0;
      top: 0;
      color: #fff;
      width: 14px;
      height: 14px;
      background: #D9D9D9;
      padding: 2px;
      font-size: 9px;
    }
  }

  .ant-table-row {
    &:hover {
      .ant-table-cell {
        border-right: 1px solid #f0f0f0;
      }
    }
  }

  .ant-table-thead>tr>th {
    background-color: #F2F2F2;
  }

  .checklist-num {
    font-size: 14px;
    color: #3296FA;
    cursor: pointer;
  }
}

// 查看图片的弹框样式
.report-wrap-modal {
  .img-list {
    display: flex;
    flex-wrap: wrap;
    padding-left: 38px;
    .item {
      margin-right: 16px;
      margin-bottom: 16px;
      width: 120px;
      height: 90px;
      position: relative;
      &:nth-child(5n) {
        margin-right: 0;
      }

      img {
        width: 120px;
        height: 90px;
      }

      .time {
        text-align: center;
        font-size: 14px;
        color: #666666;
        width: 120px;
        height: 24px;
        line-height: 24px;
        background: #F4F4F4;
        border: 1px solid #EFEFEF;
        position: absolute;
        bottom: 0;
        left: 0;
      }
    }
  }
}
// 生成报告弹框样式
.create-report-modal {
  .report-name {
    font-size: 14px;
    display: flex;
    align-items: center;
    width: 372px;
    margin: 0 auto 155px;
    input {
      width: 300px;
      height: 40px;
      margin-left: 15px;
    }
  }
}

// 添加字段弹框
.add-field-modal {
  .ant-checkbox-group {
    width: 100%;
    padding-left: 15px;
  }
  .ant-checkbox-group-item {
    // display: inline-block;
    margin-right: 6px;
    width: 32%;
    text-align: left;
    font-size: 14px;
    margin-bottom: 20px;
  }
}
