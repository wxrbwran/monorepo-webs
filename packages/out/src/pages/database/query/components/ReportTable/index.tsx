import React, { useEffect, useState } from 'react';
import { Table, message } from 'antd';
import { CloseOutlined } from '@ant-design/icons';
import { useSelector } from 'react-redux';
import ImgViewer from '../ImgViewer';
// import full from '@/assets/img/icon_full.svg';
// import fullExit from '@/assets/img/icon_full_exit.svg';
import add from '@/assets/img/icon_add.svg';
import CreateReport from '../CreateReport';
import AddField from '../AddField';
import * as api from '@/services/api';
import { sexList } from 'xzl-web-shared/dist/utils/consts';
import { queryFields, columnFields } from '../../consts';
import { handleTableDataSource } from '../../util';
import './index.scss';
import type { ColumnsType } from 'antd/lib/table';

interface IProps {
  location: {
    pathname: string;
  };
  tableData: ITableData[];
  reportId?: string;
  refreshTable: () => void;
  handleSearchData?: () => void; // 添加字段，调用查询接口
  handleExport?: (type: string) => void;
}

function ReportTable({
  location,
  tableData,
  reportId,
  refreshTable,
  handleSearchData,
  handleExport,
}: IProps) {
  const [tableColumns, setTableColumns] = useState<ColumnsType<IDataSource>>([]); // 表格字段
  const [tableDataSource, setTableDataSource] = useState<IDataSource[]>([]); // table的数据源
  const [tableFull, _setTableFull] = useState<boolean>(false); // 表格是否放大展示
  const [changeTitle, setChangeTitle] = useState<string[]>([]); // 用户添加或删除后要展示的title
  const { baseCondition, images, other, queryScope } = useSelector((state: IQuery) => state.query);

  // 删除字段
  const handleDelField = (name: string, titles: []) => {
    // 报告页面删除字段需要调用接口
    if (location.pathname === '/database/report') {
      const params = {
        reportId,
        type: name,
        // titles: tableData.title.filter(item => item !== name)
      };
      api.report.patchReportTitle(params).then(() => {
        message.success('删除字段成功');
        refreshTable();
      });
    } else {
      // 查询页面，删除字段，是前端过滤掉不显示
      console.log('changeTitle', changeTitle);
      console.log('titles', titles);
      const newTitles = titles.filter((item) => item !== name);
      if (newTitles.length === 0) {
        message.error('请至少保留一个字段');
      } else {
        /* eslint-disable @typescript-eslint/no-use-before-define */
        initData([...newTitles]);
        setChangeTitle([...newTitles]);
      }
    }
  };

  const renderTitle = (label: string, titles: []) => {
    return (
      <div>
        {queryFields[label]}
        <CloseOutlined onClick={() => handleDelField(label, titles)} />
      </div>
    );
  };

  const renderColomm = (item: string, text: string, record: { sid: string }) => {
    switch (item) {
      case 'sex':
        return <span>{sexList[+text]}</span>;
      case 'hypertension':
      case 'hyperglycemia':
      case 'hyperlipemia':
      case 'hyperuricemia':
        return +text === 1 ? '有' : '无';
      case 'shqx':
      case 'xcg':
      case 'bcg':
      case 'xzcs':
      case 'xdt':
        return (
          <ImgViewer imageType={item} sid={record.sid}>
            {text || '0'}
          </ImgViewer>
        );
      default:
        return text;
    }
  };
  const initData = (titles: string[]) => {
    const columns: ColumnsType<IDataSource> = [];
    titles.forEach((item) => {
      columns.push({
        title: renderTitle(item, titles),
        dataIndex: item,
        key: item,
        render: (text: any, record: any) => <>{renderColomm(item, text, record)}</>,
      });
    });
    setTableColumns(columns);
  };

  useEffect(() => {
    let dataSource = [];
    let titles: string[] = [];
    if (location.pathname === '/database/report') {
      tableData.forEach((item) => {
        Object.keys(item).forEach((el) => {
          if (!item[el]) {
            /* eslint-disable no-param-reassign */
            delete item[el];
          }
        });
      });
      dataSource = tableData;
      // indexOfMax 从字段最多的obj中拼接colunm
      const lenArr = dataSource.map((item) => Object.keys(item).length);
      const max = Math.max(...lenArr);
      const indexOfMax = lenArr.indexOf(max);
      titles = Object.keys(dataSource[indexOfMax]).filter((item) => item !== 'sid');
    } else {
      dataSource = handleTableDataSource(tableData);
      // 扩展查询需要用到这个参数
      window.$storage.setItem(
        'PSidArr',
        dataSource.map((item) => item.sid),
      );
      const baseTitle = tableData[0]?.members.filter((item) => item.baseTitle)[0]?.baseTitle;
      titles = baseTitle?.map((item: string) => columnFields[item]);
    }
    setChangeTitle(titles);
    setTableDataSource(dataSource);
    initData(titles);
  }, [tableData]);

  const handleCreateReport = (name: string) => {
    let newSource: any = [];
    tableDataSource.forEach((item) => {
      let newObj = {};
      changeTitle.forEach((i) => {
        newObj = {
          ...newObj,
          [i]: item[i],
          sid: item.sid,
        };
      });
      newSource = [...newSource, newObj];
    });
    const params = {
      name,
      projectSid: window.$storage.getItem('sid'),
      condition: {
        base: [...baseCondition],
        images,
        other: other && !!other.fourHigh ? other : null,
        queryScope,
      },
      teams: [...newSource],
    };
    api.query.generatorReport(params).then(() => {
      message.success('生成报告成功');
    });
  };

  return (
    <div className={['report_wrap_table', tableFull ? 'full' : ''].join(' ')}>
      {location.pathname === '/database/report' ? (
        <div className="table-export">
          <span onClick={() => handleExport('EXCEL')}>导出Excel</span>
          <span onClick={() => handleExport('WORD')}>导出Word</span>
        </div>
      ) : (
        <div className="table_top">
          <div className="table-top__info">
            根据您的搜索条件，为您智能筛选字段。您可以对结果字段进行增加或删除
          </div>
          <div className="btn-wrap">
            <span className="create-btn mr-10">
              <CreateReport handleCreateReport={handleCreateReport}>生成报告</CreateReport>
            </span>
            {/* <span className="full-screen">
              <img
                data-testid="full-btn"
                src={tableFull ? full : fullExit}
                alt=""
                onClick={() => setTableFull(!tableFull)}
              />
            </span> */}
            {changeTitle && handleSearchData && (
              <AddField
                handleChangeTitle={initData}
                changeTitle={changeTitle}
                handleSearchData={handleSearchData}
              >
                <img src={add} alt="添加字段" />
              </AddField>
            )}
          </div>
        </div>
      )}
      <Table
        dataSource={tableDataSource}
        columns={tableColumns}
        rowKey={(record: { id: string }) => record.id}
        pagination={false}
      />
    </div>
  );
}
export default ReportTable;
