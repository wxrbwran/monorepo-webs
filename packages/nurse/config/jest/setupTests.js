// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import 'jest';
// import axios from 'axios';
// import '@testing-library/jest-dom/extend-expect';

// Enzyme.configure({ adapter: new Adapter() });

jest.mock('@/services/auth', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
  };
  return {
    ajax: API,
    auth: API,
  };
});

jest.mock('@/services/http', () => {
  const API = {
    get: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    post: jest.fn(() =>
      Promise.resolve({
        data: {},
        status: 'success',
      }),
    ),
    setAuthorizationToken: jest.fn(),
  };
  const http = API;
  return http;
});

// jest.mock('@/services/api', () => {
//     const API = {
//         get: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         post: jest.fn(() => Promise.resolve({
//             data: {}, status: 'success',
//         })),
//         setAuthorizationToken: jest.fn(),
//     };
//     const http = API;
//     return http;
// })
