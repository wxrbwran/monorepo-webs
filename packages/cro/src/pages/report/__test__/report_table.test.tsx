import 'jest';
import React from 'react';
import ReportTable from '../components/report_table';
import { render, fireEvent, RenderResult } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore({
  query: {
    images: [],
    other: [],
    base: {}
  },
  user: {
    user: {
      id: 'feqw'
    }
  }
});
const tableData = {
  name: '报告名字',
  response: {
    extData: [],
    patientBase: [
      {
        id: '232',
        age: 20,
        name: '小明'
      },
      {
        id: '222',
        age: 30,
        name: '小卷'
      },
    ],
  },
  title: ['姓名', '年龄', '身高', '体重'],
  wordUrl: '',
  excelUrl: '',
  request: {
    base: {
      maxAge: 20,
      minAge: 10,
      maxHeight: 190,
      minHeight: 100,
      maxWeight: 180,
      minWeight: 60,
      gender: 'MALE'
    },
    images: [],
    other: {
      fourHigh: []
    },
  },
}
const reportProps = {
  location: {
    pathname: '/report/detail'
  },
  tableData: tableData,
  refreshTable: jest.fn(),
  reportId: '100',
}
const queryProps = {
  location: {
    pathname: '/query/query_result'
  },
  tableData: tableData,
  refreshTable: jest.fn(),
  handleSearchData: jest.fn(),// 添加字段，调用查询接口
  sourceTitle: ['体重']
}
describe('查询————表格组件', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    wrapper = render(<Provider store={store}><ReportTable {...queryProps} /></Provider>)
  })
  it('匹配快照', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('查询页面:按钮显示是否正确', () => {
    expect(wrapper.getByText('生成报告')).toBeInTheDocument();
  });
  it('表格删除字段功能', () => {
    const tableTitle = wrapper.container.getElementsByTagName('th');
    expect(tableTitle.length).toBe(2);
    const firstClose = wrapper.container.querySelector('.ant-table-thead > tr > th > div > .anticon-close');
    firstClose && fireEvent.click(firstClose);
    expect(tableTitle.length).toBe(1);
  })
  it('全屏交互测试', () => {
    expect(wrapper.container.querySelector('.full')).toBeFalsy();
    const fullBtn = wrapper.getByTestId('full-btn');
    fireEvent.click(fullBtn);
    expect(wrapper.container.querySelector('.full')).toBeTruthy();
  });
});
describe('报告—————表格组件', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    wrapper = render(<Provider store={store}><ReportTable {...reportProps} /></Provider>)
  })
  it('报告页面,表格上方功能按钮显示是否正确', async () => {
    expect(wrapper.getByText('导出Excel')).toBeInTheDocument();
  });
  it('表格渲染正常,2条数据', async () => {
    const rowTr = wrapper.container.getElementsByClassName('ant-table-tbody')[0].children;
    expect(rowTr.length).toBe(2);
  });
})