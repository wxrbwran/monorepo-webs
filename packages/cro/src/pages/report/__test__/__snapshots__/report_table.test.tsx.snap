// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`查询————表格组件 匹配快照 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="report-wrap-table "
      >
        <div
          class="table-top"
        >
          <div
            class="table-top__info"
          >
            根据您的搜索条件，为您智能筛选字段。您可以对结果字段进行增加或删除
          </div>
          <div
            class="btn-wrap"
          >
            <span
              class="create-btn"
            >
              <span
                class="create-report"
                data-testid="plugin"
              >
                生成报告
              </span>
            </span>
            <span
              class="full-screen"
            >
              <img
                alt=""
                data-testid="full-btn"
                src="icon_full_exit.svg"
              />
            </span>
            <span>
              <img
                alt="添加字段"
                src="icon_add.svg"
              />
            </span>
          </div>
        </div>
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup />
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                          >
                            <div>
                              姓名
                              <span
                                aria-label="close"
                                class="anticon anticon-close"
                                role="img"
                                tabindex="-1"
                              >
                                <svg
                                  aria-hidden="true"
                                  class=""
                                  data-icon="close"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            <div>
                              年龄
                              <span
                                aria-label="close"
                                class="anticon anticon-close"
                                role="img"
                                tabindex="-1"
                              >
                                <svg
                                  aria-hidden="true"
                                  class=""
                                  data-icon="close"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="232"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            小明
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            20
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="222"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            小卷
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            30
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="report-wrap-table "
    >
      <div
        class="table-top"
      >
        <div
          class="table-top__info"
        >
          根据您的搜索条件，为您智能筛选字段。您可以对结果字段进行增加或删除
        </div>
        <div
          class="btn-wrap"
        >
          <span
            class="create-btn"
          >
            <span
              class="create-report"
              data-testid="plugin"
            >
              生成报告
            </span>
          </span>
          <span
            class="full-screen"
          >
            <img
              alt=""
              data-testid="full-btn"
              src="icon_full_exit.svg"
            />
          </span>
          <span>
            <img
              alt="添加字段"
              src="icon_add.svg"
            />
          </span>
        </div>
      </div>
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          <div>
                            姓名
                            <span
                              aria-label="close"
                              class="anticon anticon-close"
                              role="img"
                              tabindex="-1"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="close"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                />
                              </svg>
                            </span>
                          </div>
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          <div>
                            年龄
                            <span
                              aria-label="close"
                              class="anticon anticon-close"
                              role="img"
                              tabindex="-1"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="close"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                />
                              </svg>
                            </span>
                          </div>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="232"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          小明
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          20
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="222"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          小卷
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          30
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
